frontend/src/hooks/queryBuilder/useAutoComplete.ts

```diff
@@ -49,18 +64,18 @@ export const useAutoComplete = (query: IBuilderQuery): IAutoComplete => {
 		isFreeText,
 		handleSearch,
 		query,
+		setSearchKey,
 	);
 
 	const handleSelect = useCallback(
 		(value: string): void => {
 			if (isMulti) {
 				setSearchValue((prev: string) => {
-					if (prev.includes(value)) {
-						return prev.replace(` ${value}`, '');
-					}
-					return checkStringEndsWithSpace(prev)
-						? `${prev} ${value}`
-						: `${prev} ${value},`;
+					const matches = prev?.matchAll(TAG_FSM);
+					const [match] = matches ? Array.from(matches) : [];
+					const [, , , matchTagValue] = match;
+					const data = Papa.parse(matchTagValue).data.flat();
+					return replaceStringWithMaxLength(prev, data as string[], value);
```

Comment:
I see you got rid of some white space and replaced it with `matchAll` from the expression. What does this code do?

Reply:
This code will replace the last search string with the selected option or the user will allow a custom key. @srikanthccv 
The previous value will return all options you selected(string with comma separator) + at the ends with your searched value.
hence `replaceStringWithMaxLength` will do the same.